design/login.go

```diff
@@ -111,4 +111,26 @@ var _ = a.Resource("logout", func() {
 		a.Response(d.TemporaryRedirect)
 		a.Response(d.InternalServerError, JSONAPIErrors)
 	})
+
+	// The logoutv2 endpoint performs invalidation of all user tokens managed by the auth service, before redirecting the
+	// client to the oauth provider's logout endpoint.  To enable this, the endpoint must be
+	// secured (i.e. a valid access token is required in the Authorization header) in contrast to the unsecured, original logout endpoint
+	// which simply redirects the user to the oauth provider's logout URL
+	a.Action("logoutv2", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/v2"),
+		)
+		a.Headers(func() {
+			a.Header("Referer", d.String)
+		})
+		a.Params(func() {
+			a.Param("redirect", d.String, "URL to be redirected to after successful logout. If not set then will redirect to the referrer instead.")
+		})
+		a.Description("Logout user")
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.TemporaryRedirect)
```

Comment:
I don't think we can use TemporaryRedirect here. We have to use window.location in browser to redirect to logout endpoint to be able to handle 307 responses. But we can't add any headers to such redirects (when using window.location).
We had the same problem with out account/token linking API. As a workaround we return 200 OK with a payload with redirectLocation - https://github.com/fabric8-services/fabric8-auth/blob/master/design/token.go#L162
I think we have to do the same here. So, in UI when some user clicks on logout then we will make ajax call to our logout v2 API with the access token in the Auth header and extract the redirect location from its response payload. Then use window.location to navigate to that location.